src/utils/plotSpecs/generateFrequencySpec.js

```diff
@@ -200,6 +188,19 @@ const generateData = (hierarchy, properties, config) => {
     hierarchy.find((rootNode) => rootNode.key === key)?.children
   ));
 
+  if (config.frequencyType === 'proportional') {
+    // Get the total number of cells in each cell set.
+    cellSets.x.forEach((xCellSet, indx) => {
+      let total = 0;
+      cellSets.y.forEach((yCellSet) => {
+        const yCellSetIds = Array.from(properties[yCellSet.key].cellIds);
+        const xCellSetIds = Array.from(properties[xCellSet.key].cellIds);
+        total += xCellSetIds.filter((id) => yCellSetIds.includes(id)).length;
+      });
+      cellSets.x[indx].totalY = total;
+    });
+  }
```

Comment:
Can this block be put under the conditional check below? That way if the data is empty, we do not need to do this calculation

Reply:
If i put it under it will be calculated for every xCellSetCluster * yCellSetCluster we only need to do it for every sample

Reply:
Not inside the loop, I mean after the if check on line 204. That check checks for the presence of `cellSets.x` and `cellSets.y`.